webapi/src/main/scala/org/knora/webapi/responders/Responder.scala

```diff
@@ -153,6 +178,24 @@ abstract class Responder(responderData: ResponderData) extends LazyLogging {
       }
     } yield ()
 
+  /**
+   * Throws an exception if a class is used in data.
+   *
+   * @param classIri                  the IRI of the class.
+   * @param errorFun                  a function that throws an exception. It will be called if the class is used.
+   */
+  protected def throwIfClassIsUsedInData(
+    classIri: SmartIri,
+    errorFun: => Nothing
+  ): Future[Unit] =
+    for {
+      classIsUsed: Boolean <- isClassUsedInData(classIri)
+
+      _ = if (classIsUsed) {
+        errorFun
+      }
+    } yield ()
```

Comment:
would it be worth considering using one of scala's ways of avoiding exception throwing (like `Either`) here?

Reply:
As discussed, I won't change it here because it conforms to the overall pattern of V2.